src/MSALWrapper/MSALWrapper.csproj

```diff
@@ -38,13 +38,24 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="3.1.4" />
-    <PackageReference Include="Microsoft.Identity.Client" Version="4.43.1" />
-    <PackageReference Include="Microsoft.Identity.Client.Extensions.Msal" Version="2.19.6" />
-    <PackageReference Include="Microsoft.IdentityModel.JsonWebTokens" Version="6.16.0" />
+    <PackageReference Include="Microsoft.Identity.Client.Extensions.Msal" Version="2.24.0" />
+    <PackageReference Include="Microsoft.IdentityModel.JsonWebTokens" Version="6.23.1" />
+  </ItemGroup>
+
+  <ItemGroup Condition="$([MSBuild]::IsOSPlatform('Windows')) == false">
+    <!-- Not Windows: simple package reference of 4.47 -->
+    <PackageReference Include="Microsoft.Identity.Client" Version="4.47.1" />
   </ItemGroup>
 
-  <ItemGroup Condition=" '$(TargetFramework)' == 'net472' ">
-    <PackageReference Include="Microsoft.Identity.Client.Desktop" Version="4.43.1" />
+  <ItemGroup Condition="$([MSBuild]::IsOSPlatform('Windows'))">
+    <!-- Windows: Explicitly continue using the net5 version of MSAL -->

```

Comment:
Will you link to https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3682 here, @kyle-rader?